File: backend/TheTribe.Domain/Entities/Connection.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace TheTribe.Domain.Entities;

public enum ConnectionStatus
{
    Pending = 0,
    Accepted = 1,
    Rejected = 2,
    Blocked = 3
}

public class Connection
{
    public Guid Id { get; set; } = Guid.NewGuid();

    public Guid RequesterId { get; set; }
    public User Requester { get; set; } = null!;

    public Guid ReceiverId { get; set; }
    public User Receiver { get; set; } = null!;

    public ConnectionStatus Status { get; set; } = ConnectionStatus.Pending;

    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public DateTime? UpdatedAt { get; set; }
}

File: shared/src/types/connection.ts
export type ConnectionStatus = "pending" | "accepted" | "rejected";

export interface Connection {
    id: string;
    requesterId: string;
    receiverId: string;
    status: ConnectionStatus;
    createdAt: string;
    partner?: {
        id: string;
        firstName: string;
        lastName: string;
        profilePhotoUrl?: string | null;
    };
}
